test/server.test.js

```diff
@@ -1924,8 +1924,12 @@ test("should emit 'after' on successful request", function(t) {
     });
 
     SERVER.get('/foobar', function(req, res, next) {
-        res.send('hello world');
-        next();
+        setTimeout(function() {
+            res.send('hello world');
+            setTimeout(function() {
+                next();
+            }, 500);
```

Comment:
Why do we need any `setTimeout` call here?

Reply:
We generate some "work" in the server to be sure that events are firing in the right order. In my previous fix tests passed with positive false.

Reply:
What do you mean by "some work"? It seems it's not CPU-bound "work", so is it a way to make events on request/response objects to happen in a specific order or within a certain timeframe? If so, should we document what the test is actually testing?

And why not keeping the original test? Should be add a new test instead of replacing the existing one?

Reply:
Remaining questions above still relevant:

1. Should we document _why_ we want to control the timing of those operations in this test?
2. Should we also keep the original test and add a new test with those changes?